--- a/net/minecraft/server/level/ServerLevel.java
+++ b/net/minecraft/server/level/ServerLevel.java
@@ -1630,7 +_,7 @@
 
             return map.object2IntEntrySet()
                 .stream()
-                .sorted(Comparator.comparing(Entry::getIntValue).reversed())
+                .sorted(Comparator.comparing(o -> ((Entry<String>)o).getIntValue()).reversed())
                 .limit(5L)
                 .map(entry -> entry.getKey() + ":" + entry.getIntValue())
                 .collect(Collectors.joining(","));
