--- a/net/minecraft/server/network/ServerCommonPacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerCommonPacketListenerImpl.java
@@ -19,8 +_,13 @@
 import net.minecraft.network.protocol.common.ServerboundKeepAlivePacket;
 import net.minecraft.network.protocol.common.ServerboundPongPacket;
 import net.minecraft.network.protocol.common.ServerboundResourcePackPacket;
+import net.minecraft.network.protocol.common.custom.BrandPayload;
+import net.minecraft.network.protocol.common.custom.CustomPacketPayload;
+import net.minecraft.network.protocol.configuration.custom.ServerboundLogMessageAckPayload;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.server.level.ClientInformation;
+import net.minecraft.server.network.config.JoinWorldTask;
+import net.minecraft.server.network.config.custom.LogMessageTask;
 import net.minecraft.util.VisibleForDebug;
 import org.slf4j.Logger;
 
@@ -68,6 +_,21 @@
 
     @Override
     public void handleCustomPayload(ServerboundCustomPayloadPacket packet) {
+        
+        System.out.printf("called handleCustomPayload, packet: %s\n", packet.payload().id().toString());
+        CustomPacketPayload payload = packet.payload();
+        if (payload instanceof BrandPayload brandPayload) {
+            System.out.printf("received brand payload: %s\n", brandPayload.brand());
+        } else {
+            this.handleCustomPayload(payload);
+//            System.out.printf("received unknown payload: %s\n", payload.id().toString());
+        }
+    }
+    
+    protected abstract void handleCustomPayload(CustomPacketPayload payload);
+    
+    protected void handleUnknownCustomPayload(CustomPacketPayload payload) {
+        LOGGER.warn("Unknown custom packet payload: {}", payload.id());
     }
 
     @Override
