--- a/net/minecraft/world/level/block/PointedDripstoneBlock.java
+++ b/net/minecraft/world/level/block/PointedDripstoneBlock.java
@@ -583,8 +_,9 @@
     @Nullable
     public static BlockPos findStalactiteTipAboveCauldron(Block block, Level level, BlockPos blockPos) {
         if (block instanceof PointedDripstoneBlock pointedDripstoneBlock) {
-            BiPredicate biPredicate = (blockPos1, blockState) -> canDripThrough(level, blockPos1, blockState);
-            return findBlockVertical(level, blockPos, Direction.UP.getAxisDirection(), biPredicate, pointedDripstoneBlock::canDrip, 11).orElse(null);
+            BiPredicate biPredicate = (blockPos1, blockState) -> canDripThrough(level,
+                (BlockPos) blockPos1, (BlockState) blockState);
+            return (BlockPos) findBlockVertical(level, blockPos, Direction.UP.getAxisDirection(), biPredicate, pointedDripstoneBlock::canDrip, 11).orElse(null);
         } else {
             return null;
         }
