--- a/net/minecraft/network/protocol/common/ClientboundCustomPayloadPacket.java
+++ b/net/minecraft/network/protocol/common/ClientboundCustomPayloadPacket.java
@@ -25,6 +_,7 @@
 import net.minecraft.network.protocol.common.custom.StructuresDebugPayload;
 import net.minecraft.network.protocol.common.custom.VillageSectionsDebugPayload;
 import net.minecraft.network.protocol.common.custom.WorldGenAttemptDebugPayload;
+import net.minecraft.network.protocol.configuration.custom.ClientboundLogMessagePayload;
 import net.minecraft.resources.ResourceLocation;
 
 public record ClientboundCustomPayloadPacket(CustomPacketPayload payload) implements Packet<ClientCommonPacketListener> {
@@ -51,6 +_,7 @@
         .put(StructuresDebugPayload.ID, StructuresDebugPayload::new)
         .put(VillageSectionsDebugPayload.ID, VillageSectionsDebugPayload::new)
         .put(WorldGenAttemptDebugPayload.ID, WorldGenAttemptDebugPayload::new)
+        .put(ClientboundLogMessagePayload.ID, ClientboundLogMessagePayload::new)
         .build();
 
     public ClientboundCustomPayloadPacket(FriendlyByteBuf buffer) {
